pkg/catalogv2/oci/client.go

```diff
@@ -215,8 +224,26 @@ func (o *Client) getAuthClient() (*http.Client, error) {
 	baseTransport := http.DefaultTransport.(*http.Transport).Clone()
 	baseTransport.TLSClientConfig = config
 
+	if o.exponentialBackOffValues != nil {
+		if o.exponentialBackOffValues.MaxRetries > 0 {
+			retryPolicy.MaxRetry = o.exponentialBackOffValues.MaxRetries
+		}
+		if o.exponentialBackOffValues.MaxWait != nil {
+			retryPolicy.MaxWait = o.exponentialBackOffValues.MaxWait.Duration
+		}
+		if o.exponentialBackOffValues.MinWait != nil {
+			retryPolicy.MinWait = o.exponentialBackOffValues.MinWait.Duration
+		}
+	}
+	retryPolicy.Backoff = retry.ExponentialBackoff(retryPolicy.MinWait, 2, 0.2)
+
+	retryTransport := retry.NewTransport(baseTransport)
+	retryTransport.Policy = func() retry.Policy {
+		return &retryPolicy
+	}
```

Comment:
This could keep the reconciliation loop busy just waiting. The retry policy could be combined with the reconciler's [Result.RequeueAfter](https://github.com/kubernetes-sigs/controller-runtime/blob/v0.17.3/pkg/reconcile/reconcile.go#L36) may enable a better use of resources.

Reply:
Yes @pjbgf We would like to work on iterations and use this comment as a feedback for improvement on the next PR. 

cc @diogoasouza